### PR TITLE
JIRA:VZ-2791 sockshop and todo list mc example apps

### PR DESCRIPTION
**Pull request by jmaron99**
# Description

The artifacts and documentation for deploying the sockshop and todo list sample applications in an MC environment.

Fixes VZ-2791

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

**Comment by mgianatagh** [comment]
The file examples/multicluster/README.md contains a table of the examples under multicluster.  A link should be added for this new example.

File: examples/multicluster/sock-shop/README.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+`<VERRAZZANO_HOME>/examples/sockshop`, where `<VERRAZZANO_HOME>` is the root of the Verrazzano project.
+
+**NOTE:** All files and paths in this document are relative to
+`<VERRAZZANO_HOME>/examples/sockshop`.
+
+## Deploy the Sock Shop application
+
+1. Create a namespace for the Sock Shop application by deploying the Verrazzano project.
+   ```
+   $ kubectl apply -f verrazzano-project.yaml
```
Comment (mgianatagh) [inline examples/multicluster/sock-shop/README.md:23]: The mc hello Helidon example prefixes each kubectl command with the kubeconfig of the target cluster (e.g. KUBECONFIG=$KUBECONFIG_ADMIN).  We should probably use a similar approach for this example.

File: examples/multicluster/todo-list/README.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+`<VERRAZZANO_HOME>/examples/todo-list`.
+
+## Deploy the example application
+
+1. Create a namespace for the multicluster ToDo List example by applying the Verrazzano project file.
+   ```
+   $ kubectl apply -f verrazzano-project.yaml
+   ```
+
+1. Login to the `container-registry.oracle.com` docker registry in which the todo list application image is deployed.  You
```
Comment (rosemarymarano) [inline examples/multicluster/todo-list/README.md:29]: Login to -> Log in to (log in to (v), login (n) )
docker  -> Docker (product name always capitalized)
todo list -> ToDo List (application name)

File: examples/multicluster/todo-list/README.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+will require the updated docker config.json, containing you authentication token, for the next step.
```
Comment (rosemarymarano) [inline examples/multicluster/todo-list/README.md:30]: You will require the updated docker config.json, containing you authentication token, for the next step. -> You will need the updated Docker `config.json` file, containing your authentication token, for the next step.

File: examples/multicluster/todo-list/README.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+   ```
+   $ docker login container-registry.oracle.com 
+   ```
+1. Update the mc-docker-registry-secret.yaml file with the your registry authentication info.  Edit the file and replace the
```
Comment (rosemarymarano) [inline examples/multicluster/todo-list/README.md:34]: mc-docker-registry-secret.yaml -> `mc-docker-registry-secret.yaml` (file names must be in code font)
info -> information

**Review by mgianatagh — DISMISSED** [review]

File: examples/multicluster/todo-list/README.md [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+   ```
+   $ docker login container-registry.oracle.com 
+   ```
+1. Update the mc-docker-registry-secret.yaml file with the your registry authentication info.  Edit the file and replace the 
+`<BASE 64 ENCODED DOCKER CONFIG JSON>` with the value generated from the following command.
+   ```
+   $ cat ~/.docker/config.json | base64
+   ```
+1. Create a `docker-registry` secret to enable pulling the ToDo List example image from the registry by applying the
+mc-docker-registry-secret.yaml file.  The multicluster secret resource will generate the required secret in the todo list
```
Comment (rosemarymarano) [inline examples/multicluster/todo-list/README.md:40]: mc-docker-registry-secret.yaml -> `mc-docker-registry-secret.yaml`
todo list -> `todo-list`

File: examples/multicluster/todo-list/README.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+   ```
+   $ cat ~/.docker/config.json | base64
+   ```
+1. Create a `docker-registry` secret to enable pulling the ToDo List example image from the registry by applying the
+mc-docker-registry-secret.yaml file.  The multicluster secret resource will generate the required secret in the todo list 
+namespace.
+   ```
+   $ kubectl apply -f mc-docker-registry-secret.yaml
+   ```
+1. Create the secrets for the weblogic domain by applying the mc-weblogic-domain-secret.yaml and mc-runtime-encrypt-secret.yaml files:
```
Comment (rosemarymarano) [inline examples/multicluster/todo-list/README.md:45]: weblogic  -> WebLogic (!!!)
`mc-weblogic-domain-secret.yaml` and `mc-runtime-encrypt-secret.yaml`

File: examples/multicluster/todo-list/README.md [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+   $ kubectl apply -f todo-list-application.yaml
+   ```
+
+1. Wait for the ToDo List example application to be ready.
+   You may need to repeat this command several times before it is successful.
+   The `tododomain-adminserver` pod may take a while to be created and `Ready`.
+   ```
+   $ kubectl wait pod --for=condition=Ready tododomain-adminserver -n mc-todo-list
+   ```
+   Make sure you have set the `KUBECONFIG` environment variable to point to the managed cluster's kubernetes configuration file.
```
Comment (rosemarymarano) [inline examples/multicluster/todo-list/README.md]: kubernetes  -> Kubernetes

File: examples/multicluster/todo-list/README.md [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+     * Then, you can use a browser to access the application at `https://<yourhost.your.domain>/todo/`.
+
+   Accessing the application in a browser will open a page titled "Derek's ToDo List"
+   with an edit field and an **Add** button that allows you to add tasks.
+
+1. A variety of endpoints are available to further explore the logs, metrics, and such, associated with
+   the deployed ToDo List application.
+   Accessing them may require the following:
+   
+   * Set the `KUBECONFIG` environment variable to point to the admin cluster's kubernetes configuration file.
```
Comment (rosemarymarano) [inline examples/multicluster/todo-list/README.md]: Kubernetes

File: examples/multicluster/todo-list/README.md [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+
+   | Description | Address | Credentials |
+   | ----------- | ------- | ----------- |
+   | Kibana      | `https://[vmi-system-kibana ingress host]`     | `verrazzano`/`telemetry-password` |
+   | Grafana     | `https://[vmi-system-grafana ingress host]`    | `verrazzano`/`telemetry-password` |
+   | Prometheus  | `https://[vmi-system-prometheus ingress host]` | `verrazzano`/`telemetry-password` |
+
+## Troubleshooting
+
+1. For the following commands make sure the `KUBECONFIG` environment variable is set to the path of the managed cluster's kubernetes configuration file.
```
Comment (rosemarymarano) [inline examples/multicluster/todo-list/README.md]: commands  -> commands , (comma)
Kubernetes

**Review by rosemarymarano — COMMENTED** [review]
Some edits and comments.

File: examples/multicluster/todo-list/README.md (hunk 6 above)
Comment (jmaron99) [reply examples/multicluster/todo-list/README.md:45]: It has to be capitalized in the file name?  That's going to look strange to users

Comment (rosemarymarano) [reply examples/multicluster/todo-list/README.md:45]: Are you referring to WebLogic? I don't see it as being in the file name. I see, "Create secrets for the weblogic domain..." In a file name, `weblogic` does not have to be capitalized.

File: examples/multicluster/todo-list/mc-docker-registry-secret.yaml [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+apiVersion: clusters.verrazzano.io/v1alpha1
+kind: MultiClusterSecret
+metadata:
+  name: tododomain-repo-credentials
+  namespace: mc-todo-list
+spec:
+  template:
+    type: kubernetes.io/dockerconfigjson
+    data:
+      .dockerconfigjson: <BASE 64 ENCODED DOCKER CONFIG JSON>
```
Comment (shihchang) [inline examples/multicluster/todo-list/mc-docker-registry-secret.yaml:12]: Do we have docs or tool to generate this secret or that base64 string? 
I created a non-mc docker registry secret and then copy paste that ".dockerconfigjson:  ..."

Comment (jmaron99) [reply examples/multicluster/todo-list/mc-docker-registry-secret.yaml:12]: I put instructions on the previous step so I'm confused about what you are asking?

Comment (shihchang) [reply examples/multicluster/todo-list/mc-docker-registry-secret.yaml:12]: got it. i should have read that.

File: examples/multicluster/todo-list/README.md [hunk 11]
```diff
@@ -26,8 +26,8 @@ The ToDo List application deployment artifacts are contained in the Verrazzano p
    $ KUBECONFIG=$KUBECONFIG_ADMIN kubectl apply -f verrazzano-project.yaml
    ```
 
-1. Login to the `container-registry.oracle.com` docker registry in which the todo list application image is deployed.  You
-will require the updated docker config.json, containing you authentication token, for the next step.
+1. Log in to the `container-registry.oracle.com` Docker registry in which the todo list application image is deployed.  You
+will need the updated Docker config.json, containing you authentication token, for the next step.
```
Comment (rosemarymarano) [inline examples/multicluster/todo-list/README.md:30]: config.json, containing you -> `config.json`, containing your (file name in code font)

**Review by rosemarymarano — APPROVED** [review]
Just one left, otherwise fine.

**Review by rosemarymarano — APPROVED** [review]
LGTM

**Review by mgianatagh — APPROVED** [review]

**Pull request merged**